.github/workflows/tests.yaml

```diff
@@ -0,0 +1,44 @@
+name: tests
+on:
+  push:
+    branches:
+      - master
+      - '*.x'
+  pull_request:
+    branches:
+      - master
+      - '*.x'
+jobs:
+  tests:
+    name: ${{ matrix.name }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - {name: Linux, python: '3.8', os: ubuntu-latest, toxenv: py}
+          - {name: '3.7', python: '3.7', os: ubuntu-latest, toxenv: py}
+          - {name: '3.6', python: '3.6', os: ubuntu-latest, toxenv: py}
+          - {name: 'PyPy', python: pypy3, os: ubuntu-latest, toxenv: py}
+          - {name: Style, python: '3.8', os: ubuntu-latest, toxenv: style}
+          - {name: Docs, python: '3.8', os: ubuntu-latest, toxenv: docs}
+          - {name: Windows, python: '3.8', os: windows-latest, toxenv: py}
+          - {name: Mac, python: '3.8', os: macos-latest, toxenv: py}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+      - name: cache pip
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: pip|${{ matrix.python }}|${{ hashFiles('requirements/*.txt') }} }}
```

Comment:
i think you need to include the OS as well since you have both Windows and Linux in the matrix

Reply:
I figured only a few deps at most would have platform wheels, so it seemed fine to use one cache for both. We can make the cache key more specific if you want.